aws/resource_aws_workspace_test.go

```diff
@@ -0,0 +1,455 @@
+package aws
+
+import (
+	"fmt"
+	"log"
+	"reflect"
+	"testing"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/workspaces"
+	multierror "github.com/hashicorp/go-multierror"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/acctest"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/terraform"
+)
+
+func init() {
+	resource.AddTestSweepers("aws_workspace", &resource.Sweeper{
+		Name: "aws_workspace",
+		F:    testSweepWorkspaces,
+	})
+}
+
+func testSweepWorkspaces(region string) error {
+	client, err := sharedClientForRegion(region)
+	if err != nil {
+		return fmt.Errorf("error getting client: %w", err)
+	}
+	conn := client.(*AWSClient).workspacesconn
+
+	var errors error
+	input := &workspaces.DescribeWorkspacesInput{}
+	err = conn.DescribeWorkspacesPages(input, func(resp *workspaces.DescribeWorkspacesOutput, _ bool) bool {
+		for _, workspace := range resp.Workspaces {
+			err := workspaceDelete(aws.StringValue(workspace.WorkspaceId), conn)
+			if err != nil {
+				errors = multierror.Append(errors, err)
+			}
+
+		}
+		return true
+	})
+	if testSweepSkipSweepError(err) {
+		log.Printf("[WARN] Skipping workspaces sweep for %s: %s", region, err)
+		return errors // In case we have completed some pages, but had errors
+	}
+	if err != nil {
+		errors = multierror.Append(errors, fmt.Errorf("error listing workspaces: %s", err))
+	}
+
+	return errors
+}
+
+func TestAccAwsWorkspace_basic(t *testing.T) {
```

Comment:
Since `workspace_properties` is optional, and all of its fields are optional, this test function should be split up to test various cases:
* `TestAccAwsWorkspace_basic` should cover the case with no `workspace_properties` and importing it. It could also test adding a `workspace_properties` with values
* A test for a `workspace_properties`, but with no values set
* Tests for setting `workspace_properties` with values

Reply:
On my way...